credits.txt

```diff
@@ -61,3 +79,188 @@ Beta Testers
   Sean Fahey
   Mauricio Gomes
   Iaz Poolar
+
+Special thanks to MCOfficer for
+creating, managing, and hosting
+the nightly builds that makes
+so much of this work possible.
+
+Proof Readers & Editors
+  Arrow2thekn33
+  Bladewood
+  Fzzr
+  jafdy
+  LocalGod79
+  MCOfficer
+  Moonbeam
+  RecursiveParadox
+  Tadrix
+  temtemy
+  Zearth Goh aka TheMarksman
+
+Mission & Story Creators
+  Amazinite
+  Anarchist2
+  Arachi-Lover
+  beccaBunny
+  czartrak
+  Fzzr
+  Janaszar
+  LocalGod
+  MasterOfGrey
+  MessyMix
+  Petervdmeer
+  Pointedstick
+  UnfetteredOne
+  Zitchas
+
+Thank you to all the following
+people on GitHub who have
+contributed to Endless-Sky:
```

Comment:
Quite some people in this section are already mentioned earlier under Programming, Proof Readers or Story Creators. I would suggest not repeating them here, since they are already mentioned in more specific sections.

Reply:
I left them in this section for clarity, inclusiveness, and simplicity: 
- Some people on the specific lists may want to use an alternative name.
- The list of GitHub contributors is long, and likely to get longer.
- The longer the list is, the more likely we are to have similarly named contributors.
- If the GitHub Contributors list is a straight forward list of everyone who has contributed, then we never have to worry about trying to curate it or filter out people. (whose names may or may not actually be matches for the names in the specific lists).

I initially left the Devs out of it because they are easy to identify, but I am fond of the idea of having one big non-curated list that includes all GitHub contributors specifically so we never have to risk excluding someone by accidentally omitting them because they have a name similar to that of one of the people called out in the curated section.

Reply:
In that case this should be programmatically generated from the repo's commit history or contributor graph. I don't recall if the github display correctly handles now-deactivated accounts (i.e. @ "ghost"), so if it does not then we must parse the commit graph.

Reply:
I made the "GitHub Contributors" list the hard way, with Amazinite's help pulling to get the contributers beyond the 100th spot. If there's an automatic way to pull the list and/or update it in the future, sounds great! I don't really want to distract you and our other coder herders from doing game-specific stuff to work on something to parse the script, though. We've got a list now; it shouldn't be too hard to maintain it.

Or were you referring to catching all the people that have made commits/suggestions directly to PRs and thus aren't registered as having committed to the game itself?

Reply:
My suggestion is to use a programmatic solution such as the GitHub API (which probably won't handle "ghost") or git commands with some other language (bash, JavaScript, Python) to handle author aggregation and formatting.

Manual parsing of this information is an untenable chore.

Reply:
i tried https://github.com/pydanny/contributors and immediately ran into the ratelimit, no surprise. i could make it work, though, i'm pretty sure.

once we're at it, we can easily set up an action that creates PRs with acknowledgments for new contributors. which reminds me, that's already a thing: https://allcontributors.org/

Reply:
That sounds pretty good.

As far as that goes, how about getting the list in its current form merged for the 0.9.12 so that people can see that it has truly become a community project; and have an automatic system for adding people to it in the future be something that is developed for 0.9.13?

Reply:
agreed, moved to #4970

Reply:
welp, just as i say it, it works. oneliner to get all contributors:
```
$ hub api repos/endless-sky/endless-sky/contributors --paginate | jq '.[].login' -r 
```

here's the output for your convenience: https://gist.github.com/MCOfficer/9b4710251d8ff02a41c5dc52401ed422

Reply:
Thanks!